pytorch_lightning/callbacks/progress/rich_progress.py

```diff
@@ -368,8 +368,12 @@ def _add_task(self, total_batches: int, description: str, visible: bool = True)
                 f"[{self.theme.description}]{description}", total=total_batches, visible=visible
             )
 
-    def _update(self, progress_bar_id: int, current: int, total: Union[int, float], visible: bool = True) -> None:
-        if self.progress is not None and self._should_update(current, total):
+    def _update(self, progress_bar_id: int, current: int, visible: bool = True) -> None:
+        if self.progress is not None and self.is_enabled:
```

Comment:
You check `is_enabled` again here because we wouldn't be able to access `total`?
Because it's already checked inside `should_update`

Reply:
yes, because in rich we don't initialize tasks if it's disabled. we can remove it from `_should_update`